### PR TITLE
Baseline integration

### PR DESCRIPTION
**Pull request by Egorrko**
This PR experiment is about generating and checking errors using the baseline file.

I understand that I haven't written the tests yet. Please rate this PR as an idea.

### CODE REVIEW

**Comment by jsh9** [comment]
Hi @Egorrko , I think this is a cool idea.  Please feel free to keep experimenting on this branch.

**Comment by Egorrko** [comment]
@jsh9 I think this PR is ready for your viewing and testing

**Comment by jsh9** [comment]
Hi @Egorrko , would you mind writing some instructions on how to use this feature?  You can reply here and I'll put them into README myself.  Thanks!

**Comment by Egorrko** [comment]
Sorry about my CI linting problems
I can't reproduce this tox environment to check all this problems myself

File: pydoclint/main.py [hunk 1]
```diff
@@ -209,6 +237,7 @@ def validateStyleValue(
     ),
     is_eager=True,
     callback=injectDefaultOptionsFromUserSpecifiedTomlFilePath,
+    default='pyproject.toml',
```
Comment (Egorrko) [inline pydoclint/main.py:240]: Look at this change:
I think it could be useful for most of users

**Comment by jsh9** [comment]
> Sorry about my CI linting problems I can't reproduce this tox environment to check all this problems myself

That's fine. I know what those CI errors are from and I can fix them for you.

What I don't quite know is: as a user, how would I use the new feature that you are adding here.

File: pydoclint/main.py (hunk 1 above)
Comment (jsh9) [reply pydoclint/main.py:240]: Hey, I'm sorry but I don't quite understand your comment here.

**Comment by Egorrko** [comment]
@jsh9
How to use the baseline parameter:
First, you need to generate a baseline file.
Use the `--generate-baseline True` key, and the `--baseline <PATH>` key (or add the string `baseline ="pydoclint-baseline.txt"` to your config file).
After that, your file will be created, and it will contain all the current errors in the project and you will be able to include it in the current CI.
All old errors will be ignored, and any new error will be displayed.
You need to run pydoclint as usual - in check mode.

**Comment by Egorrko** [comment]
Look at this examples:
https://flakeheaven.readthedocs.io/en/latest/commands/baseline.html
https://github.com/orsinium-labs/mypy-baseline

File: pydoclint/main.py (hunk 1 above)
Comment (Egorrko) [reply pydoclint/main.py:240]: I think that now for most of the projects to keep the `pyproject.toml` file in the root is the standard

**Comment by jsh9** [comment]
> @jsh9 How to use the baseline parameter: First, you need to generate a baseline file. Use the `--generate-baseline True` key, and the `--baseline <PATH>` key (or add the string `baseline ="pydoclint-baseline.txt"` to your config file). After that, your file will be created, and it will contain all the current errors in the project and you will be able to include it in the current CI. All old errors will be ignored, and any new error will be displayed. You need to run pydoclint as usual - in check mode.

Thanks!

I'll try it out tomorrow and then I'll write these instructions into README and publish a new version.

**Comment by RastaDill** [comment]
Yea. Thats cool idea. @jsh9 did you manage to view the instructions?

**Comment by jsh9** [comment]
Hi @RastaDill and @Egorrko , I haven't had a chance to get to it these past few days -- sorry!  Swamped by other things.  I'll have a chance to get to it tomrorow.

**Comment by jsh9** [comment]
Hi @Egorrko , sorry I was still too busy to actually test it and write README.  If you'd like to contribute to this PR, please feel free to do so.  Thank you!

**Comment by jsh9** [comment]
Hi @Egorrko , thanks so much for adding the documentation.  I've read through them and made some tweaks.  I just merged the MR and I'll publish the new version now.

**Comment by jsh9** [comment]
0.3.3 published: https://github.com/jsh9/pydoclint/releases/tag/0.3.3

**Pull request merged**